### PR TITLE
Change Modal esc listener to keydown from keyup

### PR DESCRIPTION
**Pull request by jeffredodd**
This PR changes the non-standard functionality of the esc key listener in the modal from `keyup` to `keydown` and fixes issue #194. 

The main issue can be reproduced by embedding a DropdownButton inside of a modal. When using keyboard navigation inside the dropdown button, attempting to close the menu with escape will also close the modal itself which creates a jarring user experience.

### CODE REVIEW

File: src/Modal.js [hunk 1]
```diff
@@ -94,6 +94,12 @@ class Modal extends React.Component {
     /**
      * A callback fired when the escape key, if specified in `keyboard`, is pressed.
      */
+    onEscapeKeyDown: PropTypes.func,
+
+    /**
+     * Support for this function will be depricated. Please use `onEscapeKeyDown` instead
```
Comment (jquense) [inline src/Modal.js]: "deprecated" also can you add the `@deprecated` doclet in the comment (under the description)

File: src/Modal.js [hunk 2]
```diff
@@ -428,11 +434,14 @@ class Modal extends React.Component {
     }
   }
 
-  handleDocumentKeyUp = (e) => {
+  handleDocumentKeyDown = (e) => {
     if (this.props.keyboard && e.keyCode === 27 && this.isTopModal()) {
-      if (this.props.onEscapeKeyUp) {
-        this.props.onEscapeKeyUp(e);
+      let escapeFunc = this.props.onEscapeKeyDown || this.props.onEscapeKeyUp;
```
Comment (jquense) [inline src/Modal.js]: this is a bit misleading. I'd make sure the old handler actually gets called on keyUp

File: src/Modal.js (hunk 1 above)
Comment (jeffredodd) [inline src/Modal.js]: let me know if my recent change is the correct way to declare deprecated

**Comment by jquense** [comment]
lgtm

File: src/Modal.js [hunk 3]
```diff
@@ -94,6 +94,13 @@ class Modal extends React.Component {
...
      * A callback fired when the escape key, if specified in `keyboard`, is pressed.
      */
+    onEscapeKeyDown: PropTypes.func,
+
+    /**
+     * Support for this function will be deprecated. Please use `onEscapeKeyDown` instead
+     * A callback fired when the escape key, if specified in `keyboard`, is pressed.
+     * @deprecated
+     */
     onEscapeKeyUp: PropTypes.func,
```
Comment (taion) [inline src/Modal.js]: use https://github.com/react-bootstrap/prop-types-extra#deprecatedvalidator-reason maybe? that way people will get the warning; makes it easier to remove for next version

we're already pulling in `prop-types-extra`

File: src/Modal.js [hunk 4]
```diff
@@ -428,12 +440,21 @@ class Modal extends React.Component {
     }
   }
 
+  handleDocumentKeyDown = (e) => {
+    if (this.props.keyboard && e.keyCode === 27 && this.isTopModal()) {
```
Comment (taion) [inline src/Modal.js]: can we do `e.key === 'Escape'`? (and below too)

i know this is an old issue, but it'd clean things up a tiny bit

**Comment by taion** [comment]
Thanks!

**Pull request merged**